smartsim/entity/entity.py

```diff
@@ -107,17 +107,16 @@ def __init__(self, name: str) -> None:
         :param name: Name of the entity
         """
         self.name = name
+        """The name of the application"""
+
+    @abc.abstractmethod
```

Comment:
Is there a better name for ``as_program_arguments``?  I'm strugglign to think of the perfect wording... maybe ``as_sequence`` or ``as_executable_sequence``?